library/src/scala/quoted/Expr.scala

```diff
@@ -52,6 +52,16 @@ object Expr {
   def unapply[T](x: Expr[T])(using FromExpr[T])(using Quotes): Option[T] =
     scala.Predef.summon[FromExpr[T]].unapply(x)
 
+  /** Creates an expression that will construct a function with one parameter
+   *
+   *  Transforms the expression
+   *    `a => r` where `a: Expr[A]` and `r: Expr[R]`
+   *  to an expression equivalent to
+   *    `'{ $a => $r }` typed as an `Expr[A => R]`
+   */
+  def ofFunction1[A, R](f: Expr[A] => Expr[R])(using Type[R], Type[A])(using Quotes): Expr[A => R] =
```

Comment:
If we add this we should add it for all function arities. It seems that we would need `TupledFunction` (see #10416) to be able to write this interface for all function sizes.